### ISSUE TITLE
Dashboards

### ISSUE THREAD

**Issue by dkumor** [heedy/heedy#361]
Currently, views happen either at a per-timeseries, or as a per-query form. Heedy should gain support for generic dashboarding functionality to show multiple plots/pieces of data from multiple different queries, or even from multiple different object types (not just timeseries - for example, a notebook view could display a plot generated in a python notebook).

- [x] Backend
- [ ] Frontend

**Comment by mammuth** [comment]
I like the notebook views idea :+1: 

Another braindump: To me, flexible visualization is a core feature/requirement of a tool like heedy. Doing visualization and data discoverability "nicely" sounds non-trivial, though, but there are many mature tools that excel at this task, eg. Grafana.

Wouldn't it make sense to leverage that and - as a first step - add heedy as a data-source for Grafana? :thinking: 

One downside is obviously that it's not "one central tool" anymore, but one might be able to do it similarly like the jupyter integration in the frontend?

**Comment by dkumor** [comment]
Yeah, Grafana is a powerful tool for analysis, and writing code to allow Grafana to take the data stored in Heedy and visualize it would be really cool! I think this would be an extension to Grafana -  I would welcome contributions in that area!

What you said about customizability is extremely important - and is the current focus. I am now updating Heedy to allow users to arbitrarily modify/add visualizations for timeseries with custom javascript (since a visualization needs to decide when to show itself based on the data it gets). 

Once that is done, I will add the dashboarding functionality (which is currently half-finished). Dashboards will allow not only displaying custom plots, but also taking arbitrary jupyter notebook outputs and displaying those - and having them auto-update too!

I expect the custom visualizations for timeseries to be ready mid-summer, and to have alpha-quality dashboards end of summer.
